onnxscript/tests/test_dort.py

```diff
@@ -0,0 +1,472 @@
+        op_level_debug=False,
+    )
+    # Convert the exported result to ONNX ModelProto.
+    onnx_model = exported.to_model_proto(opset_version=target_opset)
+
+    # Modify ONNX model using pre-registered graph transforms.
+    # They are in-place modifications for avoiding unnecessary
+    # copy of ONNX initializers.
+    return onnx_model
+
+
+def custom_backend(
+    graph_module: "torch.fx.GraphModule",  # noqa: F821
+    args: List["torch.Tensor"],  # noqa: F821
+    target_opset: Optional[int] = None,
+    verbose: Union[int, Tuple[int, int]] = 0,
+    use_cuda: bool = False,
+) -> Callable:
+    import onnxruntime
+    import torch
+
+    onx = _dynamo_export(graph_module, args, verbose, target_opset)
+
+    providers = ["CPUExecutionProvider"]
+    if use_cuda:
+        providers.insert(0, "CUDAExecutionProvider")
+    sess = onnxruntime.InferenceSession(
+        onx.SerializeToString(), providers=["CPUExecutionProvider"]
+    )
+    input_names = [i.name for i in sess.get_inputs()]
+
+    def run(*args, **kwargs):
+        assert len(kwargs) == 0, f"Not implemented with kwargs={kwargs}"
+        # All inputs are moved to cpu with numpy. Not efficient but it simplifies
+        # the unit tests.
+        feeds = dict(zip(input_names, map(lambda x: x.detach().cpu().numpy(), args)))
+        results = sess.run(None, feeds)
+        return tuple(map(torch.Tensor, results))
+
+    return run
+
+
+def get_decomposition_table():
+    import torch
+
+    new_table = {}
+    for k, v in torch._decomp.decomposition_table.items():
+        if k.name() in {"aten::sigmoid_backward"}:
+            new_table[k] = v
+    return new_table
+
+
+class TestCustomOps(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        import onnxruntime  # noqa: F401
+
+    def test_llama(self):
+        import torch
+        import torch.onnx
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            from transformers import LlamaConfig
+            from transformers.models.llama.modeling_llama import LlamaModel
+
+        local_aot_ort, _ = make_aot_ort(
+            dynamic=False,
+            rewrite=True,
+            verbose=1,
+        )
+
+        config = LlamaConfig(
+            hidden_size=16,
+            num_hidden_layers=1,
+            vocab_size=1024,
+            intermediate_size=16,
+            max_position_embeddings=1024,
+            num_attention_heads=2,
+        )
+        config._attn_implementation = "eager"
+
+        model = LlamaModel(config)  # .to("cuda")
+
+        batch, seq, vocab_size = 2, 1024, 1024
+        input_ids = ids_tensor([batch, seq], vocab_size)  # .to("cuda")
+        # input_mask = torch.tril(torch.ones(batch, seq, dtype=torch.float32))
+
+        model(input_ids)  # , input_mask)
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            optimized_mod = torch.compile(model, backend=local_aot_ort, fullgraph=True)
```

Comment:
This looks like testing the custom backend "local_aot_ort" rather than dort, which is "onnxrt" backend. Do you have more context?

Reply:
I guess this is to inject rewriter ...

Reply:
It comes from here: https://github.com/pytorch/pytorch/blob/main/test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py#L31. It is also used to specify options such as inject the rewriter without any code change in pytorch.